tests/integration/high_availability/test_self_healing.py

```diff
@@ -194,3 +194,62 @@ async def test_freeze_db_process(ops_test: OpsTest, continuous_writes) -> None:
     ), "The deployed mysql application does not have three online nodes"
 
     await ensure_all_units_continuous_writes_incrementing(ops_test)
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.self_healing_tests
+async def test_graceful_crash_of_priamry(ops_test: OpsTest, continuous_writes) -> None:
```

Comment:
nit: 
```suggestion
async def test_graceful_crash_of_primary(ops_test: OpsTest, continuous_writes) -> None:
```

Reply:
addressed in [fc75968](https://github.com/canonical/mysql-k8s-operator/pull/112/commits/fc759680f8d5ac7b94e46fdcfa6fa8fd4810863b)